From 93c1cf305dd677b9ecc0f9e2c91cd6ee4ab51bd9 Mon Sep 17 00:00:00 2001
From: Ludovic Desroches <ludovic.desroches@atmel.com>
Date: Thu, 24 Jan 2013 16:16:15 +0100
Subject: [PATCH 296/319] video: atmel_hlcdfb: now, suspend/resume is managed
 only by the base layer

When going to suspend or resume the functions will be called by the base layer
but also by the overlay layers. It's not necessary to do it several time.
Moreover the base address for overlays is not the base address of the IP so
offsets to access some registers are wrong.

Signed-off-by: Ludovic Desroches <ludovic.desroches@atmel.com>
---
 drivers/video/atmel_hlcdfb.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/video/atmel_hlcdfb.c b/drivers/video/atmel_hlcdfb.c
index 3897c9b..d1136a5 100644
--- a/drivers/video/atmel_hlcdfb.c
+++ b/drivers/video/atmel_hlcdfb.c
@@ -423,9 +423,13 @@ static irqreturn_t atmel_hlcdfb_interrupt(int irq, void *dev_id)
 
 static int atmel_hlcdfb_suspend(struct platform_device *pdev, pm_message_t mesg)
 {
+	const struct platform_device_id *id = platform_get_device_id(pdev);
 	struct fb_info *info = platform_get_drvdata(pdev);
 	struct atmel_lcdfb_info *sinfo = info->par;
 
+	if (strcmp(id->name, "atmel_hlcdfb_base"))
+		return 0;
+
 	/*
 	 * We don't want to handle interrupts while the clock is
 	 * stopped. It may take forever.
@@ -444,9 +448,13 @@ static int atmel_hlcdfb_suspend(struct platform_device *pdev, pm_message_t mesg)
 
 static int atmel_hlcdfb_resume(struct platform_device *pdev)
 {
+	const struct platform_device_id *id = platform_get_device_id(pdev);
 	struct fb_info *info = platform_get_drvdata(pdev);
 	struct atmel_lcdfb_info *sinfo = info->par;
 
+	if (strcmp(id->name, "atmel_hlcdfb_base"))
+		return 0;
+
 	atmel_lcdfb_start_clock(sinfo);
 	atmel_hlcdfb_start(sinfo);
 	if (sinfo->atmel_lcdfb_power_control)
-- 
1.7.10.4

